analyses/malawi/dryspells_trigger/exploration/compare_arc2_chirps.Rmd

```diff
@@ -0,0 +1,483 @@
+  ))
+
+arc2_chirps_lj2 <- fuzzy_full_join(
+  arc2_overlap_df,
+  chirps_df,
+  match_fun = list(`==`, `>=`, `<=`),
+  by = c(
+    "pcode" = "pcode",
+    "arc2_center_end" = "chirps_start",
+    "arc2_center_end" = "chirps_end"
+  ))
+
+arc2_chirps_lj3 <- fuzzy_full_join(
+  arc2_overlap_df,
+  chirps_df,
+  match_fun = list(`==`, `>=`, `<=`),
+  by = c(
+    "pcode" = "pcode",
+    "arc2_touch_start" = "chirps_start",
+    "arc2_touch_start" = "chirps_end"
+  ))
+
+arc2_chirps_lj4 <- fuzzy_full_join(
+  arc2_overlap_df,
+  chirps_df,
+  match_fun = list(`==`, `>=`, `<=`),
+  by = c(
+    "pcode" = "pcode",
+    "arc2_touch_end" = "chirps_start",
+    "arc2_touch_end" = "chirps_end"
+  ))
+
+arc2_chirps_df <- bind_rows(arc2_chirps_lj1,
+                            arc2_chirps_lj2,
+                            arc2_chirps_lj3,
+                            arc2_chirps_lj4) %>%
+  distinct() %>%
+  mutate(pcode.x = ifelse(is.na(pcode.x), pcode.y, pcode.x)) %>%
+  rename(pcode = pcode.x) %>%
+  select(-pcode.y) %>% 
+  # clean up duplicate joined rows
+  group_by(pcode, arc2_center_start, arc2_touch_start) %>%
+  filter(!(n() > 1 & is.na(chirps_start))) %>%
+  group_by(pcode, arc2_center_start, chirps_start) %>%
+  filter(!(n() > 1 & is.na(arc2_touch_start))) %>%
+  group_by(pcode, chirps_start, arc2_touch_start) %>%
+  filter(!(n() > 1 & is.na(arc2_center_start))) %>%
+  group_by(pcode, chirps_start) %>%
+  filter(!(n() > 1 & is.na(arc2_center_start) & !is.na(chirps_start))) %>%
+  ungroup() %>%
+  select(-starts_with("ADM2_EN"))
+
+# get simple ADM2_EN data set
+adm2_df <- map(list(
+  arc2_center_df,
+  arc2_touch_df,
+  chirps_df
+), ~ select(.x, pcode, ADM2_EN)) %>%
+  bind_rows() %>%
+  distinct()
+
+# add ADM2_EN to main data frame
+arc2_chirps_df <- arc2_chirps_df %>%
+  left_join(adm2_df, by = "pcode")
+```
+
+### Simple comparison
+
+At a basic level, we detect about 1/3 more dry spells using ARC2 data than using CHIRPS data,
+with `r nrow(arc2_center_df)` and `r nrow(arc2_touch_df)` detected with the centroid and touching methods
+respectively, and only `r nrow(chirps_df)` using CHIRPS. Let's compare how and when
+dry spells are detected by each method.
+
+```{r detection-matrix}
+arc2_chirps_df <- mutate(arc2_chirps_df,
+  chirps_ds = !is.na(chirps_confirmation),
+  arc2_centroid_ds = !is.na(arc2_center_confirmation),
+  arc2_touch_ds = !is.na(arc2_touch_confirmation)
+)
+
+arc2_chirps_df %>%
+  select(CHIRPS = chirps_ds,
+         `ARC2 (Centroid)` = arc2_centroid_ds,
+         `ARC2 (Touching)` = arc2_touch_ds) %>%
+  group_by_all() %>%
+  summarize(Occurrences = n(),
+            .groups = "drop") %>%
+  mutate(across(-Occurrences, ~ ifelse(.x, "Detected", "Not detected"))) %>%
+  arrange(desc(Occurrences)) %>%
+  knitr::kable()
+```
+
+We can see above a breakdown of dry spell detection by method. Interestingly, where
+CHIRPS detects a dry spells, both ARC2 methods either also detect one or don't.
```

Comment:
So this would be confirmation that both ARC2 methods are consistent with each other?

Reply:
Yeah, at least in their difference with CHIRPS. Basically used that to say I was comfortable just comparing ARC2 centroid to CHIRPS following from there.